### ISSUE TITLE
JSON dumps crashes if hdparm output contains non-ASCII characters

### ISSUE THREAD

**Issue by chris-iot** [thomas-krenn/TKperf#27]
During testing a disk me and my collegues encountered that some disks do have non-ASCII characters in the output of hdparm. Here is an excerpt, the full output is attached:

```
ATA device, with non-removable media
	Model Number:       VRFEM5240GCCAMTL 
	Serial Number:      V00000000023        
	Firmware Revision:  R0706A  
	Media Serial Num:   ü
	Transport:          Serial, ATA8-AST, SATA 1.0a, SATA II Extensions, SATA Rev 2.5, SATA Rev 2.6, SATA Rev 3.0
```

The resulting stacktrace is:
```
ich@ich-desktop:~$ sudo tkperf ssd Vtest /dev/sdb -nj 1 -iod 4 -ssdt iops
[sudo] password for ich:
!!!Attention!!!
All data on /dev/sdb will be lost!
Are you sure you want to continue? (In case you really know what you are doing.)
Press 'y' to continue, any key to stop:
y
Starting ssd mode...
Testing device:
Model Number: VRFEM5240GCCAMTL
Serial Number: V00000000023
Firmware Revision: R0706A
00
Media Serial Num: 03􀀁
device size with M = 1000*1000: 240057 MBytes (240 GB)
write-caching = 1 (on)
Starting test: iops
Traceback (most recent call last):
File "/usr/local/bin/tkperf", line 149, in <module>
myTest.run()
File "/usr/local/lib/python2.7/dist-packages/perfTest/PerfTest.py", line 291, in
run
self.toXml()
File "/usr/local/lib/python2.7/dist-packages/perfTest/PerfTest.py", line 199, in
toXml
self.getDevice().toXml(e)
File "/usr/local/lib/python2.7/dist-packages/perfTest/Devices.py", line 315, in
toXml
data = json.dumps(self.__devinfo)
File "/usr/lib/python2.7/json/__init__.py", line 244, in dumps
return _default_encoder.encode(obj)
File "/usr/lib/python2.7/json/encoder.py", line 201, in encode
return encode_basestring_ascii(o)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xfc in position 156: invalid
start byte
```

The "Media serial number" is not needed in my/our case, so it can be omitted completely. Another approach is to replace non-ASCII characters with a default ASCII character or remove the non-ASCII characters. This would also work if the non-ASCII characters appear in fields that are needed like e.g. the Model number.

During today's meeting we discussed this issue with @gschoenberger so he already knows about it and what to do next.
[non-ascii.txt](https://github.com/thomas-krenn/TKperf/files/2425036/non-ascii.txt)

**Comment by gschoenberger** [comment]
Commit 1fdaf9df6b78a088dc90d34ca6016a62dde8d773 checks if __deviceinfo contains non ASCII charactars before calling json.dumps.
Still need to test the commit...

**Comment by gschoenberger** [comment]
This issue is resolved with commit adb00354c4e72c6efd60f0cff8f3d7f6619f2d26
As xml requires only unicode/ASCII chars, lines with non ASCII chars are skipped while fetching device information.
